packages/playwright-core/src/server/android/android.ts

```diff
@@ -256,15 +260,23 @@ export class AndroidDevice extends SdkObject {
     await androidBrowser._init();
     this._browserConnections.add(androidBrowser);
 
+    const artifactsDir = await fs.promises.mkdtemp(ARTIFACTS_FOLDER);
+    socket.on('close', async () => {
```

Comment:
- Maybe `androidBrowser.onclose = () => { ... }` is better?
- You could also add to  "gracefully close set" to cleanup on process exit like this: https://github.com/microsoft/playwright/blob/71edad327b7bd07bf00eea510a1edd5e258048bf/packages/playwright-core/src/server/chromium/chromium.ts#L192-L194

Reply:
`androidBrowser.onclose` gets used internally by the transport, gets set there so we can't override it. Added it to the gracefully close set for safety, this gives us additional protection when running e.g. in the driver mode.